src/common/http/http.cc

```diff
@@ -53,6 +53,23 @@ bool IsFormDataSafeCharacter(const char character) {
   return IsUrlSafeCharacter(character) || (character == '+');
 }
 
+// We found that at least accounts.google.com[1] could contain `/` in the
+// authorization code, as part of OAuth callback URl query param.
+// such as `..../?code=4/Abc38.
+// RFC 6749[2] specifies "code" could be `VSCHAR`, containing all printable
+// ascii chars. RFC 3986[3] about URI specifies that:
+// If a reserved character is found in a URI component and
+//  no delimiting role is known for that character, then it must be
+//  interpreted as representing the data octet corresponding to that
+//  character's encoding in US-ASCII.
+// [1]
+// https://developers.google.com/identity/protocols/oauth2/openid-connect#sendauthrequest
+// [2] https://datatracker.ietf.org/doc/html/rfc6749#appendix-A.11.
+// [3] https://www.rfc-editor.org/rfc/rfc3986#page-12
+bool IsOIDCCodeSafeCharacter(const char character) {
```

Comment:
I think that `http.cc` should be independent from filter implementation. It seems this rule is OIDC specific. Thus we should move this to `oidc_filter.cc` or new file such as `oidc_query_decoder.cc` and pass it to `Http::DecodeQueryData`. For example, I prefer to implement that as follows:

```
class UrlQueryDecoder {
  virtual absl::string_view decode(absl::string_view) = 0;
};
```

```
class OidcUrlQueryDecoder : public UrlQueryDecoder {
    // implement...
};
```

But, we should prioritize to fix this problem, Therefore we should postpone this refactor as TODO.

Reply:
Add a TODO.

I think putting out of http lib is fine (this is not filter lib, but http, btw). but I'm not sure if we have to setup class/inheritance, especially if there's only one method, without actual state to maintain.
Pure util function seems work well and clear.

Reply:
ping, is that sufficient for now? @Shikugawa

Reply:
Yes. SGTM.